### PR TITLE
Develop Post class & test case (SWEET-13)

### PR DESCRIPTION
**Pull request by JeonJinil**
SWEET-13

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 .gradle
-.idea
\ No newline at end of file
+.idea
+
+build
+/build
```
Comment (hj-d) [inline .gitignore]: Please add new line.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -10,7 +10,7 @@ spring.jpa.database=mysql
 
 spring.datasource.url=jdbc:mysql://localhost:3306/STUDY_DB?useSSL=false&characterEncoding=UTF-8&serverTimezone=Asia/Seoul&allowPublicKeyRetrieval=true&useSSL=false
 spring.datasource.username=root
-spring.datasource.password=fndlqhtm4132
+spring.datasource.password=root123123
```
Comment (hj-d) [inline src/main/resources/application.properties:13]: I think... this is your password.
right??
So we should fix username and password.
```
spring.datasource.username=root
spring.datasource.password=password
```

File: src/main/resources/static/static/index.html [hunk 3]
```diff
@@ -1 +1 @@
...
...out((function(){u({type:"timeout",target:a})}),12e4);a.onerror=a.onload=u,document.head.appendChild(a)}return Promise.all(r)},i.m=e,i.c=n,i.d=function(e,r,t){i.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},i.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},i.t=function(e,r){if(1&r&&(e=i(e)),8&r)return e;if(4&r&&"object"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(i.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:e}),2&r&&"string"!=typeof e)for(var n in e)i.d(t,n,function(r){return e[r]}.bind(null,n));return t},i.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return i.d(r,"a",r),r},i.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},i.p="/",i.oe=function(e){throw console.error(e),e};var a=this.webpackJsonpfrontend=this.webpackJsonpfrontend||[],c=a.push.bind(a);a.push=r,a=a.slice();for(var f=0;f<a.length;f++)r(a[f]);var l=c;t()}([])</script><script src="/static/js/2.f0f4e7bf.chunk.js"></script><script src="/static/js/main.67e977d2.chunk.js"></script></body></html>
```
Comment (hj-d) [inline src/main/resources/static/static/index.html:1]: Please add new line.

File: src/main/resources/static/static/static/js/2.f0f4e7bf.chunk.js.map [hunk 4]
```diff
@@ -0,0 +1 @@
...EAAEwnB,WAAW,OAAa,IAAItV,EAAE/sB,WAAW+sB,EAAEyqC,aAAa,qBAAwBzqC,EAAE,IAAI,IAAI/T,EAAEA,EAAE6B,EAAE4nB,WAAW5nB,EAAE3K,YAAY8I,GAAG,OAAO,IAAIk+C,GAAGr8C,EAAE,EAAEkS,EAAE,CAAC6d,SAAQ,QAAI,GAC3B6sB,CAAGz+C,EAAEkgB,GAAGC,EAAEnG,EAAEqkC,cAAiB,oBAAoBjjC,EAAE,CAAC,IAAIn1B,EAAEm1B,EAAEA,EAAE,WAAW,IAAIvZ,EAAEk8C,GAAG59B,GAAGl6B,EAAEd,KAAK0c,IAAIq6C,IAAG,WAAW4B,GAAG/pC,EAAEoM,EAAEte,EAAEuZ,MAAK,OAAO2iC,GAAG59B,GAGlG,SAASu+B,GAAG78C,EAAEkS,GAAG,IAAI/T,EAAE,EAAEnb,UAAUC,aAAQ,IAASD,UAAU,GAAGA,UAAU,GAAG,KAAK,IAAIy5D,GAAGvqC,GAAG,MAAMtE,MAAMvpB,EAAE,MAAM,OAAO23D,GAAGh8C,EAAEkS,EAAE,KAAK/T,GA1BtW+5C,GAAG,SAASl4C,EAAEkS,EAAE/T,GAAG,IAAIkgB,EAAEnM,EAAEo2B,MAAM,GAAG,OAAOtoC,EAAE,GAAGA,EAAE8sC,gBAAgB56B,EAAEo7B,cAAc3uB,GAAEnZ,QAAQ+iC,IAAG,MAAQ,IAAG,KAAKpqC,EAAEkgB,GAAoC,CAAO,OAANkqB,IAAG,EAAUr2B,EAAEwS,KAAK,KAAK,EAAE+tB,GAAGvgC,GAAGw7B,KAAK,MAAM,KAAK,EAAEf,GAAGz6B,GAAG,MAAM,KAAK,EAAE+yB,GAAG/yB,EAAE9oB,OAAOk8C,GAAGpzB,GAAG,MAAM,KAAK,EAAEs6B,GAAGt6B,EAAEA,EAAE6Z,UAAUiE,eAAe,MAAM,KAAK,GAAG3R,EAAEnM,EAAE46B,cAAcplD,MAAM,IAAI6xB,EAAErH,EAAE9oB,KAAKi3B,SAASpC,GAAE0pB,GAAGpuB,EAAEyG,eAAezG,EAAEyG,cAAc3B,EAAE,MAAM,KAAK,GAAG,GAAG,OAAOnM,EAAE0b,cAAe,OAAG,KAAKzvB,EAAE+T,EAAE8b,MAAMka,YAAmB8K,GAAGhzC,EAAEkS,EAAE/T
```
Comment (hj-d) [inline src/main/resources/static/static/static/js/2.f0f4e7bf.chunk.js.map:1]: Should you upload chunk.js.map?

File: src/test/java/com/sweetwith/dailynote/service/posts/PostServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+public class PostServiceTest {
+
+    @Autowired
+    PostRepository postRepository;
+
+    @Autowired
+    PostService postService;
+
+    @Test
+    @DisplayName("POST 등록하고 가져와서 정확한지 확인.")
```
Comment (hj-d) [inline src/test/java/com/sweetwith/dailynote/service/posts/PostServiceTest.java:31]: How about switching to English?

File: src/test/java/com/sweetwith/dailynote/service/postlike/PostLikeServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+@Transactional
+public class PostLikeServiceTest {
+    @Autowired PostLikeRepository postLikeRepository;
+    @Autowired PostLikeService postLikeService;
+    @Autowired PostRepository postRepository;
+    @Autowired PostService postService;
+
+
+    @Test
+    @DisplayName("PostLike 등록후 조회")
```
Comment (hj-d) [inline src/test/java/com/sweetwith/dailynote/service/postlike/PostLikeServiceTest.java:30]: How about switching to English?

File: src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        // If the user has already clicked Like.
+        if(isAlreadyLike(postId,userId))
+            deletePostLike(postId);
+        else
+            registerLike(postId);
+    }
+
+    public boolean isAlreadyLike(Long postId, Long userId){
+        // TODO
+        // PostId와 UserId로 Search 해서 있으면 true, 아니면 false
```
Comment (hj-d) [inline src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java:57]: How about switching to English?

File: src/main/java/com/sweetwith/dailynote/domain/posts/PostRepository.java [hunk 8]
```diff
@@ -18,9 +18,9 @@
     List<Post> findAll();
 
     // update
-    @Modifying
+    @Modifying(clearAutomatically = true)
     @Query("UPDATE Post as p SET p.title =?2 , p.content =?3 WHERE p.Id =?1")
-    boolean updateTitleAndContent(Long id, String title, String content);
+    int updateTitleAndContent(Long id, String title, String content);
```
Comment (hj-d) [inline src/main/java/com/sweetwith/dailynote/domain/posts/PostRepository.java:23]: Why did you change the data type to int?

File: src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+public class PostLikeService {
+    private PostLikeRepository postLikeRepository;
+
+    @Autowired
+    public PostLikeService(PostLikeRepository postLikeRepository) {
+        this.postLikeRepository = postLikeRepository;
+    }
+
+    public void registerLike(Long postId) {
+        Post post = new Post();
```
Comment (hj-d) [inline src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java:23]: Why create a post here?

File: src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        postLike.setPost(post);
+
+        postLikeRepository.save(postLike);
+    }
+
+    public void deletePostLike(Long postId){
+        postLikeRepository.deleteById(postId);
+    }
+
+    public List<PostLike> getPostLikeByPostId(Long postId) {
```
Comment (kcassia) [inline src/main/java/com/sweetwith/dailynote/service/posts/PostLikeService.java:36]: Why is this return value List? Id is unique. So, I think getPostLikeByPostId is unique, too.

**Comment by hj-d** [comment]
Please do the work at the your new branch. not master..

`JeonJinil wants to merge 1 commit into SweetWith:master from JeonJinil:master`

File: src/main/java/com/sweetwith/dailynote/domain/posts/Post.java [hunk 11]
```diff
@@ -2,15 +2,16 @@
```
Comment (hj-d) [inline src/main/java/com/sweetwith/dailynote/domain/posts/Post.java:2]: I think.. you should modify folder name...

`Posts -> post`

File: src/main/resources/application.properties (hunk 2 above)
Comment (JeonJinil) [reply src/main/resources/application.properties:13]: right. It is my password. 
As if it was uploaded because of git ignore file not working.

File: src/test/java/com/sweetwith/dailynote/service/postlike/PostLikeServiceTest.java (hunk 6 above)
Comment (JeonJinil) [reply src/test/java/com/sweetwith/dailynote/service/postlike/PostLikeServiceTest.java:30]: DisplayName is widely used in Korean

File: src/test/java/com/sweetwith/dailynote/service/posts/PostServiceTest.java (hunk 5 above)
Comment (JeonJinil) [reply src/test/java/com/sweetwith/dailynote/service/posts/PostServiceTest.java:31]: DisplayName is widely used in Korean

File: src/main/java/com/sweetwith/dailynote/domain/posts/PostRepository.java (hunk 8 above)
Comment (JeonJinil) [reply src/main/java/com/sweetwith/dailynote/domain/posts/PostRepository.java:23]: JPA update method only support Integer type return.

File: src/main/java/com/sweetwith/dailynote/domain/posts/Post.java [hunk 12]
```diff
@@ -2,15 +2,16 @@
...
+import lombok.Data;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import org.apache.catalina.User;
 
 import javax.persistence.*;
 
-@Getter
-@NoArgsConstructor
+@Data
```
Comment (Wook-je) [inline src/main/java/com/sweetwith/dailynote/domain/posts/Post.java:12]: Please change the annotation. There is unnecessary information.
@data -> @@Getter, @NoArgsConstructor

File: src/main/java/com/sweetwith/dailynote/domain/posts/Post.java [hunk 13]
```diff
@@ -22,17 +23,17 @@
...
 
-    private String author;
+    private Long userId;
 
-    @Builder
-    public Posts(String title, String content, String author) {
-        this.title = title;
-        this.content = content;
-        this.author = author;
+    public Post() {
```
Comment (Wook-je) [inline src/main/java/com/sweetwith/dailynote/domain/posts/Post.java:28]: Please delete it. add @NoArgsConstructor

**Review by kcassia — COMMENTED** [review]
Review getPostLikeByPostId return type

File: src/main/java/com/sweetwith/dailynote/domain/posts/Post.java [hunk 14]
```diff
@@ -22,17 +23,17 @@
...
     }
 
-    public void update(String title, String content) {
+    @Builder
+    public Post(String title, String content, Long userId) {
         this.title = title;
         this.content = content;
+        this.userId = userId;
     }
+
```
Comment (Wook-je) [inline src/main/java/com/sweetwith/dailynote/domain/posts/Post.java:37]: Please remove the space in the code.
